### ISSUE TITLE
Invalid outdated SAN certificate delivered with mixed SAN/non-SAN ACME certificates

### ISSUE THREAD

**Issue by andig** [containous/traefik#4477]
### What did you do?

Running 1.6.4 on docker I've setup various domains using letsencrypt, at some point using SAN certifcates. Accessing primary site using https generates non-deterministic certificate errors on various platforms.

### What did you expect to see?

Lately, traefik has started to *sometimes* serve expired certificates (serial number different from the serial of the "normal" certificates). Upgrading to 1.7.8 did not help. When that happens the error persists for some time across client platforms before it vanishes again without admin action.

### What did you see instead?

It seems as if- with SANs no longer being configured- traefik was sometimes serving the meanwhile expired and non-renewed SAN certificate.

At any rate I would expect the behaviour of choosing the certficate to serve to be deterministic. Some more guard against misconfiguration might be good.

### Output of `traefik version`: (_What version of Traefik are you using?_)

````
Version:      v1.7.8
Codename:     maroilles
Go version:   go1.11.5
Built:        2019-01-29_04:33:36PM
OS/Arch:      linux/amd64
````

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

```toml
    [acme]
    email = "..."
    storage = "/etc/traefik/acme.json"
    entryPoint = "https"
    onHostRule = true
    acmeLogging = true

    [acme.httpChallenge]
    entryPoint = "http"

    [[acme.domains]]
    main = "primary"

    [[acme.domains]]
    main = "other1"

    [[acme.domains]]
    main = "other2"
```

While my current config does not contain SAN certfificates, it seems that I had them configured at one point as they are found in the `acme.json`:

```json
    {
      "Domain": {
        "Main": "primary",
        "SANs": null
      },
      "Certificate": "LS..=",
      "Key": "LS.."
    },
    {
      "Domain": {
        "Main": "primary",
        "SANs": [
          "other1",
          "other2"
        ]
      },
      "Certificate": "LS0...=",
      "Key": "LS0..."
    },
    {
      "Domain": {
        "Main": "primary",
        "SANs": [
          "other3",
          "other2"
        ]
      },
      "Certificate": "LS0...==",
      "Key": "LS0..."
    },
```

**Comment by traefiker** [comment]
Thanks for opening this issue!

We need further information to better understand the problem you're facing :thinking:

Could you please join us on our [Slack workspace](https://slack.traefik.io) and reach out to us on the ([#support channel](https://traefik.slack.com/messages/C0CDT22PJ/))?

We're looking forward to talking to you there!

**Comment by andig** [comment]
I've posted on slack:

I'm using traefik with Letsencrypt for 3 domains. I used to have the primary domain plus alternative names configured as SAN but have some time ago split the config into individual domains (multiple `[[acme.domains]]` entries).

Recently I've noticed traefik to sporadically deliver the meanwhile expired SAN certificate instead of the current single-domain certs.

Root cause seems to be that all (SAN and individual) certificates are in the `acme.json`. What I don't understand is a) why traefik chooses a cert that does not match the current configuration and b) why it only does so randomly.
